src/clj/api/server.clj

```diff
@@ -140,6 +141,34 @@
       slurp
       edn/read-string))
 
+(defn load-maps-files
+  "Load all maps files from resources."
+  []
+  {:countries-maps (into [] (.listFiles (io/file (io/resource "maps/countries"))))
+   :world-maps     (into [] (.listFiles (io/file (io/resource "maps/world"))))})
+
+(defn ensure-maps-uploaded
+  "For every map file in resources check if it is already in our s3 bucket.
+  Upload it if not."
+  [aws-config s3 bucket-name]
+  (try
+   (let [{:keys [countries-maps world-maps]} (load-maps-files)
+         maps-files (into countries-maps world-maps)]
+     (doseq [map-file maps-files]
+       (let [map-rel-path (-> map-file str (string/split #"resources/") second)
+             map-url (str "http://" (aws-s3/build-url aws-config bucket-name map-rel-path))]
+         (when-not (shared-utils/http-file-exists? map-url)
+           (log/info (str "Map file " map-url " doesn't exist, uploading..."))
+           (aws-s3/upload-file s3 {:bucket bucket-name
+                                   :key map-rel-path
+                                   :file-path (str map-file)})))))
+   (catch Exception e
+     ;; This is because I couldn't figure out how to get a dir listing of all maps in resources when
```

Comment:
I implemented this once, reading from a jar archive:https://github.com/phylogeography/SpreaD3/blob/master/src/renderers/d3/D3Renderer.java#L36
Maybe helpfull, if not, yeha we can have this one-off manual step in deployment, no kittens should die.

Reply:
Yeah this is not about reading the jar name. Is about reading the contents of a folder inside your jar resources. I tried a bunch of things but couldn't make it work. It should be possible but didn't want to spend more time on that since is nice to have but not a lot of value now.

Reply:
I wasn;t clear in my comment: the Java code below that line does exactly that, it unpacks the content of a jar. Look at the java.util.jar.JarEntry API